### PR TITLE
NMS-8824:  Add Ping Sweep RPC module

### PR DESCRIPTION
**Pull request by cgorantla**
* JIRA: http://issues.opennms.org/browse/NMS-8824
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS1048

### CODE REVIEW

File: opennms-icmp/commands/pom.xml [hunk 1]
```diff
@@ -41,5 +41,9 @@
       <groupId>org.apache.karaf.shell</groupId>
       <artifactId>org.apache.karaf.shell.console</artifactId>
     </dependency>
+	<dependency>
```
Comment (j-white) [inline opennms-icmp/commands/pom.xml]: Is this still necessary?

File: opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingCommand.java [hunk 2]
```diff
@@ -52,10 +52,19 @@
...
 
     @Option(name = "-l", aliases = "--location", description = "Location", required = false, multiValued = false)
-    String m_location = "localhost";
+    String m_location = "Default";
 
     @Option (name="-c", aliases = "--count", description="number of requests")
     int m_count = 1;
+    
+    @Option (name="-r", aliases = "--retries", description="number of retries")
+    int m_retries;
```
Comment (j-white) [inline opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingCommand.java]: Let's use these, or remove them.

File: opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package org.opennms.netmgt.icmp.commands;
```
Comment (j-white) [inline opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java]: Missing copyright header.

File: opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    int m_retries;
+
+    @Option(name = "-t", aliases = "--timeout", description = "timeout in msec")
+    int m_timeout;
+
+    @Option(name = "-p", aliases = "--packetsize", description = "packet size")
+    int m_packetsize;
+
+    @Argument(index = 0, name = "begin", description = "begin address of the IP range to be pinged", required = true, multiValued = false)
+    String begin;
```
Comment (j-white) [inline opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java]: Some variables are prefixed with m_, others are not, let's be consistent.

File: features/discovery/src/main/java/org/opennms/netmgt/discovery/actors/Discoverer.java [hunk 5]
```diff
@@ -66,13 +73,29 @@
...
 
     private final PingerFactory m_pingerFactory;
+    
+    private LocationAwarePingSweepClient client;
 
     public Discoverer(final PingerFactory pingerFactory) {
         m_pingerFactory = Preconditions.checkNotNull(pingerFactory, "pingerFactory argument");
     }
 
     public DiscoveryResults discover(DiscoveryJob job) {
```
Comment (j-white) [inline features/discovery/src/main/java/org/opennms/netmgt/discovery/actors/Discoverer.java]: Discovery will need more changes if we want to properly incorporate the PingSweep RPC client, simply changing it here is not the right approach.

File: opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java (hunk 3 above)
Comment (cgorantla) [inline opennms-icmp/commands/src/main/java/org/opennms/netmgt/icmp/commands/PingSweepCommand.java]: Done

**Pull request merged**